src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,177 @@
+
+    public static void greetLine() {
+        System.out.println("How may I be of service today?");
+    }
+
+    public static void echo() {
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+        while (!line.equals("bye")) {
+            System.out.println(line);
+            line = in.nextLine();
+        }
+    }
+
+    public static void exitLine() {
+        System.out.println("Glad I could be of help!");
+    }
+
+    public static void addList() {
+        ArrayList<Task> list = new ArrayList<>();
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+        while (!line.equals("bye")) { // condition to shut down program
+            if (line.equals("list")) { // users wants to know all text so far
+                printCurrentList(list);
+            } else if (line.startsWith("mark")) {
+                mark(line, list);
+                dukeDataStorage(arraylistToStringConverter(list));
+            } else if (line.startsWith("unmark")) {
+                unmark(line, list);
+                dukeDataStorage(arraylistToStringConverter(list));
+            } else if (line.startsWith("delete")) {
+                delete(line, list);
+                dukeDataStorage(arraylistToStringConverter(list));
+            } else {// new tasks keyed in by user
+                try {
+                    Task newTask = new Task(line);
+                    if (line.startsWith("todo")) {
+                        newTask = new ToDos(line);
+                        list.add(newTask);
+                    } else if (line.startsWith("deadline")) {
+                        newTask = new Deadlines(line);
+                        list.add(newTask);
+                    } else if (line.startsWith("event")) {
+                        newTask = new Events(line);
+                        list.add(newTask);
+                    } else {
+                        throw new NullCommandException();
+                    }
+                    dukeDataStorage(arraylistToStringConverter(list));
+                } catch (EmptyToDoException e) {
+                    System.out.println("Sire, you have yet to tell me what is it you want to do.");
+                } catch (EmptyDeadlineException e) {
+                    System.out.println("Sire, what is it that is due your specified time?");
+                } catch (EmptyEventsException e) {
+                    System.out.println("Sire, your event is unclear. Please specify.");
+                } catch (StringIndexOutOfBoundsException e) {
+                    System.out.println("Don't hold back Sire. I am here to serve.");
+                } catch (NullCommandException e) {
+                    System.out.println("Sire, I am not trained to understand gibberish.");
+                }
+            }
+            line = in.nextLine(); // read in next line of text
+        }
+    }
```

Comment:
Try to have less that 30 LOC for any method. You can split the contents of this method into multiple methods.